### PR TITLE
Upgrade to graphql-java 17.0

### PR DESCRIPTION
**Pull request by berngp**
Upgrade to [graphql-java 17.0](https://github.com/graphql-java/graphql-java/releases/tag/v17.0).

### CODE REVIEW

**Comment by EugenMayer** [comment]
also related issue is/might be #585

**Comment by setchy** [comment]
Should now be unblocked by federation-jvm 0.7.0

File: graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/LocalDateTimeScalar.kt [hunk 1]
```diff
@@ -25,19 +25,19 @@ import java.time.format.DateTimeFormatter
 
 @DgsScalar(name = "DateTime")
 class LocalDateTimeScalar : Coercing<LocalDateTime, String> {
-    override fun parseValue(input: Any?): LocalDateTime {
```
Comment (srinivasankavitha) [inline graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/LocalDateTimeScalar.kt:28]: So our kotlin users will need to make the same update? So this is a breaking change?

**Review by srinivasankavitha — APPROVED** [review]

**Review by paulbakker — APPROVED** [review]

File: graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/LocalDateTimeScalar.kt (hunk 1 above)
Comment (berngp) [reply graphql-dgs/src/test/kotlin/com/netflix/graphql/dgs/LocalDateTimeScalar.kt:28]: Yes, those who implement _Scalars_ in Kotlin will break.

**Pull request merged**